distributed/tests/test_scheduler.py

```diff
@@ -245,6 +246,43 @@ def random(**kwargs):
     test_decide_worker_coschedule_order_neighbors_()
 
 
+@pytest.mark.slow
+@gen_cluster(
+    client=True,
+    nthreads=[("127.0.0.1", 2)] * 2,
+    worker_kwargs={"memory_limit": "1.0GiB"},
+    Worker=Nanny,
+    scheduler_kwargs=dict(  # TODO remove
+        dashboard=True,
+        dashboard_address=":8787",
+    ),
+    config={
+        "distributed.worker.memory.target": False,
+        "distributed.worker.memory.spill": False,
+        "distributed.scheduler.work-stealing": False,
+    },
+)
+async def test_root_task_overproduction(c, s, *nannies):
```

Comment:
This simple test fails on main with a `KilledWorkerError` after the root tasks run a worker out of memory and restart it 3 times.